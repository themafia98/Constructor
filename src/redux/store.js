import {createStore} from 'redux';
import combineReducers from './reducers';

let broswerPluginDev = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const store = createStore(combineReducers,broswerPluginDev);
export default store;