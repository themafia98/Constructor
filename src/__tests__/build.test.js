import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';
import firebase from '../Firebase/Firebase';
import store from '../redux/testReduxStore';
import Build from '../Pages/Build/Build';



const BuildComponent = renderer.create(
    <MemoryRouter initialEntries = {['/Cabinet/Build/0']}>
        <Route path = "/Cabinet/Build/:param" >
            <Provider store = {store}>
                <Build firebase = {firebase} match = {{params: {param: 0}}} />
            </Provider>
        </Route>
    </MemoryRouter>
);

test("Build test", function(){

    let snapshotBuild = BuildComponent.toJSON();
    expect(snapshotBuild).toMatchSnapshot();

    let info = BuildComponent.root.find(item => item.props.className === 'newSectionTool CompoentnsMenu_button');
    info.props.onClick({stopPropagation: () => null});

    snapshotBuild = BuildComponent.toJSON();
    expect(snapshotBuild).toMatchSnapshot();

    let logout = BuildComponent.root.find(item => item.props.className === 'logoutButton');
    logout.props.onClick({stopPropagation: () => null});

    snapshotBuild = BuildComponent.toJSON();
    expect(snapshotBuild).toMatchSnapshot();

    let about = BuildComponent.root.find(item => item.props.className === 'about');
    about.props.onClick({stopPropagation: () => null});

    snapshotBuild = BuildComponent.toJSON();
    expect(snapshotBuild).toMatchSnapshot();
});