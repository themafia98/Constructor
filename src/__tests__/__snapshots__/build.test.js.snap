// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build test 1`] = `
<div
  onTouchStart={[Function]}
  onWheel={[Function]}
>
  <header>
    <div
      className="container"
    >
      <div
        className="flex-row"
      >
        <div
          className="header__CabinetInfo"
          onClick={[Function]}
        >
          <img
            alt="icon"
            className="mainIcon"
            draggable={false}
            onClick={null}
            src="/img/logo.svg"
            title={null}
          />
          <h3
            className="tilteApp"
          >
            Constructor
          </h3>
        </div>
        <div
          className="Navigator"
        >
          <i
            className=" about fas fa-2x fa-book"
            onClick={[Function]}
            title="Guide"
          />
          <img
            alt="icon"
            className="about"
            draggable={false}
            onClick={[Function]}
            src="/img/about_logo.svg"
            title="about"
          />
          <img
            alt="icon"
            className="logoutButton"
            draggable={false}
            onClick={[Function]}
            src="/img/logOut.svg"
            title="logOut"
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="Build"
  >
    <div
      className="sectionMode"
    >
      <button
        className="newSectionTool CompoentnsMenu_button"
        onClick={[Function]}
      >
        <span>
          New section
        </span>
      </button>
    </div>
    0
  </div>
</div>
`;

exports[`Build test 2`] = `
<div
  onTouchStart={[Function]}
  onWheel={[Function]}
>
  <header>
    <div
      className="container"
    >
      <div
        className="flex-row"
      >
        <div
          className="header__CabinetInfo"
          onClick={[Function]}
        >
          <img
            alt="icon"
            className="mainIcon"
            draggable={false}
            onClick={null}
            src="/img/logo.svg"
            title={null}
          />
          <h3
            className="tilteApp"
          >
            Constructor
          </h3>
        </div>
        <div
          className="Navigator"
        >
          <i
            className=" about fas fa-2x fa-book"
            onClick={[Function]}
            title="Guide"
          />
          <img
            alt="icon"
            className="about"
            draggable={false}
            onClick={[Function]}
            src="/img/about_logo.svg"
            title="about"
          />
          <img
            alt="icon"
            className="logoutButton"
            draggable={false}
            onClick={[Function]}
            src="/img/logOut.svg"
            title="logOut"
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="Build"
  >
    <div
      className="sectionMode"
    >
      <button
        className="newSectionTool CompoentnsMenu_button"
        onClick={[Function]}
      >
        <span>
          New section
        </span>
      </button>
    </div>
    0
  </div>
</div>
`;

exports[`Build test 3`] = `
<div
  onTouchStart={[Function]}
  onWheel={[Function]}
>
  <header>
    <div
      className="container"
    >
      <div
        className="flex-row"
      >
        <div
          className="header__CabinetInfo"
          onClick={[Function]}
        >
          <img
            alt="icon"
            className="mainIcon"
            draggable={false}
            onClick={null}
            src="/img/logo.svg"
            title={null}
          />
          <h3
            className="tilteApp"
          >
            Constructor
          </h3>
        </div>
        <div
          className="Navigator"
        >
          <i
            className=" about fas fa-2x fa-book"
            onClick={[Function]}
            title="Guide"
          />
          <img
            alt="icon"
            className="about"
            draggable={false}
            onClick={[Function]}
            src="/img/about_logo.svg"
            title="about"
          />
          <img
            alt="icon"
            className="logoutButton"
            draggable={false}
            onClick={[Function]}
            src="/img/logOut.svg"
            title="logOut"
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="Build"
  >
    <div
      className="sectionMode"
    >
      <button
        className="newSectionTool CompoentnsMenu_button"
        onClick={[Function]}
      >
        <span>
          New section
        </span>
      </button>
    </div>
    0
  </div>
</div>
`;

exports[`Build test 4`] = `
<div
  onTouchStart={[Function]}
  onWheel={[Function]}
>
  <div
    className="Build"
  >
    <div
      className="sectionMode"
    >
      <button
        className="newSectionTool CompoentnsMenu_button"
        onClick={[Function]}
      >
        <span>
          New section
        </span>
      </button>
    </div>
    0
  </div>
</div>
`;
